File: _legacy/controller/articles/delete_item.php
<?php
namespace Core;
use Exception, Controller\Main;
use Controller\files\UtilForFiles;

if (!defined('__API_GOOSE__')) exit();

/**
 * delete article
 *
 * @var Goose|Connect $this
 */

try
{
  // check and set srl
  if (($srl = (int)($this->params['srl'] ?? 0)) <= 0)
	{
    throw new Exception(Message::make('error.notFound', 'srl'));
	}

  // connect db
  $this->model->connect();

	// check access
	$token = Main::checkAccessItem($this, (object)[
		'table' => 'articles',
		'srl' => $srl,
	]);

  // get item
  $article = $this->model->getItem((object)[
    'table' => 'articles',
    'where' => 'srl='.$srl,
    'json_field' => [ 'json' ],
  ])->data;

	// remove thumbnail image
  UtilForFiles::removeFileByPath($article->json->thumbnail->path ?? '');

	// remove files
  UtilForFiles::removeAttachFiles($this, $srl, 'articles');

	// remove item
	$output = Main::delete($this, (object)[
		'table' => 'articles',
		'srl' => $srl,
	]);

  // remove comments
  $this->model->delete((object)[
    'table' => 'comments',
    'where' => 'article_srl='.$srl,
  ]);

  // set output
	if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

	// output data
	return Output::result($output);
}
catch (Exception $e)
{
  if ($this->model ?? false) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/json/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from ..file import __libs__ as file_libs
from ..tag import __libs__ as tag_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _check_token = True):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        if _check_token: checking_token(req, db)

        # set where
        where = [ f'srl = {params.srl}' ]

        # check item
        count = db.get_count(
            table_name = Table.JSON.value,
            where = where,
        )
        if count == 0: raise Exception('no data', 204)

        # delete item
        db.delete_item(
            table_name = Table.JSON.value,
            where = where,
        )

        # delete files
        file_libs.delete(db, file_libs.Module.JSON, params.srl)

        # delete tag
        tag_libs.delete(db, tag_libs.Module.JSON, params.srl)

        # set result
        result = output.success({
            'message': 'Complete delete JSON.',
        })
    except Exception as e:
        result = output.exc(e)
    finally:
        if not _db and db: db.disconnect()
        return result
